app/models/assignment.rb

```diff
@@ -752,11 +752,7 @@ def get_num_valid
 
   def get_num_marked(ta_id = nil)
     if ta_id.nil?
-      results_join = groupings.left_outer_joins(:current_result)
-      num_incomplete = results_join.where('results.id': nil)
-                                   .or(results_join.where('results.marking_state': 'incomplete'))
-                                   .count
-      get_num_assigned - num_incomplete
+      groupings.left_outer_joins(:current_result).where('results.marking_state': 'complete').count
```

Comment:
You actually need to use `joins` here, not `left_outer_joins`. This was a change I noted earlier, since now we only want to count groupings that have a result.

You can read more about the difference at https://guides.rubyonrails.org/active_record_querying.html#left-outer-joins.